viewer/web/daily.js

```diff
@@ -188,15 +191,27 @@ function submit_opinion() {
   }
 
   var new_notes = document.getElementById("opinion-notes").value
+  var decision_count = document.getElementById("opinion-count").value
 
   if (decision == new_decision && notes == new_notes) {
     // Nothing changed
     set_flash("info", "No change.")
     return
   }
 
+  // Update the count of total votes
+  if (decision == new_decision) {
+    // do nothing
+  } else if (decision == "none") {
+    // User has added a vote
+    decision_count = ++decision_count
+  } else if (new_decision == "none") {
+    // User has removed a vote
+    decision_count = --decision_count
+  }
+
   // Assemble the POST data
-  var post_data = "csd=" + csd + "&decision=" + new_decision + "&notes=" + encodeURIComponent(new_notes).replace(/%20/g, "+")
+  var post_data = "csd=" + csd + "&decision=" + new_decision + "&decision_count=" + decision_count + "&notes=" + encodeURIComponent(new_notes).replace(/%20/g, "+")
```

Comment:
I don't think you need any of this.  

There's no need to send the updated decision count back to the server here, because the server isn't going to use it.  After updating the database, the server needs to get the new count from the database again, and then send _that_ to the client.

Reply:
i.e. at the end of `update_opinion`, after the upsert, you need to do a new query to fetch the updated count of opinions, since it may have changed externally, and then save it in, say, `result["opinion_count"]` before it gets sent back to the client, something along the lines of:
```
result["opinion_count"] = DataFlagOpinion.select().where(
                DataFlagOpinion.revision == rev,
                DataFlagOpinion.type == OPINION_TYPE,
                DataFlagOpinion.lsd == csd,
).count()
```

Then here, in the load callback, add a third line to update the `opinions` JS array:
```
  if (index != -1) {
          opinions[index][0] = result.decision
          opinions[index][1] = result.notes
          opinions[index][3] = results.opinion_count  //<-- update opinion count in the array here
          console.log("opinion: " + opinions[index])
        }
```